src/components/ha-sidebar.html

```diff
@@ -102,19 +102,19 @@
     <paper-menu attr-for-selected='data-panel' selected='[[hass.currentPanel]]' on-iron-select='menuSelect'>
       <paper-icon-item on-tap='menuClicked' data-panel='states'>
         <iron-icon item-icon icon='mdi:apps'></iron-icon>
-        <span class='item-text'>States</span>
+        <span class='item-text'>[[tr(pg, "panel.states")]]</span>
       </paper-icon-item>
 
       <template is='dom-repeat' items='[[panels]]'>
         <paper-icon-item on-tap='menuClicked' data-panel$='[[item.url_path]]'>
           <iron-icon item-icon icon='[[item.icon]]'></iron-icon>
-          <span class='item-text'>[[item.title]]</span>
+          <span class='item-text'>[[tr(pg, item.title)]]</span>
```

Comment:
I guess we could have the app provide translations (or paths to ones) when they register their panel in the frontend: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/frontend/__init__.py#L72

And if we go down that path, we might even consider to remove the option to provide an actual `sidebar_title`, we will automatically generate a key based on the component name.